src/resolve/resolved-dns-packet.c

```diff
@@ -1672,6 +1697,38 @@ static bool loc_size_ok(uint8_t size) {
         return m <= 9 && e <= 9 && (m > 0 || e == 0);
 }
 
+static bool dns_svc_param_is_valid(DnsSvcParam *i) {
+        switch (i->key) {
+        /* RFC 9460, section 7.1.1: alpn-ids must exactly fill SvcParamValue */
+        case DNS_SVC_PARAM_KEY_ALPN: {
+                size_t sz = 0;
+                if (i->length <= 0)
+                        return false;
+                while (sz < i->length)
+                        sz += 1 + i->value[sz];
```

Comment:
so the values you add up here are bytes, so it's not really possible that this sum will overflow size_t. But i think an overflow check would still be nice. (or at least a comment)